src/vos/vos_io.c

```diff
@@ -37,6 +37,8 @@
 /** I/O context */
 struct vos_io_context {
 	daos_epoch_t		 ic_epoch;
+	daos_unit_oid_t		 ic_oid;
+	daos_handle_t		 ic_coh;
```

Comment:
why don't we just take the refcount on the container instead of this?

Reply:
I'll look into it in next patch

> btw, I think Fanyong said we can remove DTX_ST_INIT after changing this right? Could you see if it is true, or we have more cases for DTX_ST_INIT?

I was going to defer to Fanyong but I'll look into it